### PR TITLE
add focus-trap to popover

### PR DESCRIPTION
**Pull request by gwyneplaine**
Our popovers aren't keyboard friendly.
Previously we circumvented this in the admin-ui filter popover by autofocusing the select component. 

This is a more robust fix, which adds focus-trap to our popover component. 
I've opted for `focus-trap` over `focus-trap-react` here, as it:
* gives us more granular control 
* doesn't activate the focus-trap automatically on mount 
* doesn't pull in an unnecessary react-wrapper

Porting this over to use focus-trap-react later is arbitrary. 

Before: 

https://user-images.githubusercontent.com/12119389/117753502-eb09fb00-b25b-11eb-886c-b0902e0c9aa2.mov

After: 

https://user-images.githubusercontent.com/12119389/117753511-ef361880-b25b-11eb-933e-5db68cbc26bb.mov

### CODE REVIEW

File: design-system/packages/popover/src/Popover.tsx [hunk 1]
```diff
@@ -194,6 +196,24 @@ type DialogProps = {
 export const PopoverDialog = forwardRef<HTMLDivElement, DialogProps>(
   ({ isVisible, children, arrow, ...props }, consumerRef) => {
     const { elevation, radii, shadow, colors } = useTheme();
+    const focusTrapRef = useRef<HTMLDivElement | null>(null);
```
Comment (gwyneplaine) [inline design-system/packages/popover/src/Popover.tsx:199]: popper's provided ref is a function, and not a refObject, we create a refObject here for us to attach our focustrap to

File: design-system/packages/popover/src/Popover.tsx [hunk 2]
```diff
@@ -210,9 +230,10 @@ export const PopoverDialog = forwardRef<HTMLDivElement, DialogProps>(
           }}
           {...props}
         >
-          <div data-popper-arrow ref={arrow.ref} className="tooltipArrow" {...arrow.props} />
-
-          {children}
+          <div ref={focusTrapRef}>
```
Comment (gwyneplaine) [inline design-system/packages/popover/src/Popover.tsx:233]: Wrapping div to attach focus-trap to

File: design-system/packages/popover/src/Popover.tsx [hunk 3]
```diff
@@ -194,6 +196,24 @@ type DialogProps = {
...
+
+    useEffect(() => {
+      if (focusTrapRef?.current) {
+        focusTrap.current = focusTrapModule.createFocusTrap(focusTrapRef.current);
+      }
+    }, [focusTrapRef]);
+
+    useEffect(() => {
+      if (focusTrap?.current) {
+        if (isVisible) {
```
Comment (gwyneplaine) [inline design-system/packages/popover/src/Popover.tsx:210]: hook into isVisible to activate / deactivate the focustrap

**Review by molomby — APPROVED** [review]
LGTM

**Pull request merged**